offchain/eth-input-reader/src/eth_input_reader.rs

```diff
@@ -49,10 +58,23 @@ pub async fn start(
             .await
             .context(BrokerSnafu)?;
 
+    trace!("Creating env");
+    let env = create_env(
+        &config,
+        Arc::clone(&provider),
+        Arc::clone(&block_subscriber.block_archive),
+    )
+    .await?;
+
     trace!("Creating context");
-    let mut context =
-        create_context(&config, &state_server, &broker, dapp_metadata, metrics)
-            .await?;
+    let mut context = create_context(
```

Comment:
This made sense when this service was doing two things (sending tx and reading state). I think the role of `context` may not make sense anymore.

Could you investigate whether it still makes sense? Maybe it should be part of the `machine_driver`.

Reply:
I investigated, and I think you're right. However, the change would be more trouble than it's worth, considering this is the last update to release the reader mode and will be deprecated eventually.